pandas/core/dtypes/cast.py

```diff
@@ -556,6 +557,13 @@ def ensure_dtype_can_hold_na(dtype: DtypeObj) -> DtypeObj:
     return dtype
 
 
+_canonical_nans = {
+    np.datetime64: np.datetime64("NaT", "ns"),
```

Comment:
Will it always be correct to assume `"ns"`?

Reply:
updated so as to retain identical fill_value in object-result cases